chromium_src/chrome/browser/ui/webui/sanitized_image_source.h

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_CHROMIUM_SRC_CHROME_BROWSER_UI_WEBUI_SANITIZED_IMAGE_SOURCE_H_
+#define BRAVE_CHROMIUM_SRC_CHROME_BROWSER_UI_WEBUI_SANITIZED_IMAGE_SOURCE_H_
+
+// This juggling is required so we can still friend the Brave implementation
+// without clashing with the
+// #define SanitizedImageSource SanitizedImageSource_Chromium
+class SanitizedImageSource;
+using SanitizedImageSource_BraveImpl = SanitizedImageSource;
+
+// We override SanitizedImageSource to add support for unpadding images from the
+// Brave Private CDN.
+#define SanitizedImageSource SanitizedImageSource_Chromium
+#define OnImageLoaded                    \
+  OnImageLoaded_Chromium() {}            \
+  friend SanitizedImageSource_BraveImpl; \
+  virtual void OnImageLoaded
+
+#include "src/chrome/browser/ui/webui/sanitized_image_source.h"  // IWYU pragma: export
+
+#undef OnImageLoaded
+#undef SanitizedImageSource
+
+class SanitizedImageSource : public SanitizedImageSource_Chromium {
+ public:
+  explicit SanitizedImageSource(Profile* profile);
+
+  // For upstream tests
+  SanitizedImageSource(
+      Profile* profile,
+      scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
+      std::unique_ptr<DataDecoderDelegate> delegate);
+
+  // For Brave tests.
+  SanitizedImageSource(
+      Profile* profile,
+      scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
+      std::unique_ptr<DataDecoderDelegate> delegate,
+      std::string pcdn_domain);
```

Comment:
instead of listing these constructors explicitly try this:
```
using SanitizedImageSource_Chromium::SanitizedImageSource_Chromium;

void set_pcdn_domain_for_testing(const std::string& pcdn_domain) {
  pcdn_domain_ = pcdn_domain;
}
```
and in `SanitizedImageSource::OnImageLoaded`:
```
if (pcdn_domain_.empty()) {
  pcdn_domain_ = brave_domains::GetServicesDomain("pcdn")
}
```